### ISSUE TITLE
feature: Highlight and find rusts `todo!()` and `unimplemented!()` macro

### ISSUE THREAD

**Issue by IndianBoy42** [folke/todo-comments.nvim#186]
### Did you check the docs?

- [X] I have read all the todo-comments.nvim docs

### Is your feature request related to a problem? Please describe.

My own workflow is to use these macros a lot as stubs while planning out an API or high level algorithm. rust-analyzer also generally puts `todo!()` in generated code to satisfy the type checker. Unlike a `// TODO:` comment they aren't highlighted and findable with this plugin

### Describe the solution you'd like

Basically recognize `todo!()` and `unimplemented!()` as equivalent to a `// TODO: ` comment. Is the architecture of the plugin too tied to just searching in comments? I could try implement a PR if you think its possible.

### Describe alternatives you've considered

The alternative would be doing the highlighting myself and just using Telescope live_grep but i'd rather integrate it with this plugin.

**Comment by folke** [comment]
There's the patterns/regexes you can probably use. No idea. You'll have to figure that out on your own

**Comment by aaronkollasch** [comment]
The following options/patterns work for me (with the limitation that they may add false positives, since they are enabled for any filetype and the comment filter is disabled).

```lua
opts = {
    keywords = {
        TODO = { alt = { "todo", "unimplemented" } },
    },
    highlight = {
        pattern = {
            [[.*<(KEYWORDS)\s*:]],
            [[.*<(KEYWORDS)\s*!\(]],
        },
        comments_only = false,
    },
    search = {
        pattern = [=[\b(KEYWORDS)(:|!\()]=],
    },
},
```

`todo!()` and `unimplemented!()` get highlighted and appear in searches as expected.
